packages/edit-site/src/components/page-patterns/patterns-list.js

```diff
@@ -1,51 +1,72 @@
 /**
  * WordPress dependencies
  */
-
+import { useState, useDeferredValue, useId } from '@wordpress/element';
 import {
 	SearchControl,
-	__experimentalHeading as Heading,
-	__experimentalText as Text,
 	__experimentalVStack as VStack,
 	Flex,
 	FlexBlock,
+	__experimentalToggleGroupControl as ToggleGroupControl,
+	__experimentalToggleGroupControlOption as ToggleGroupControlOption,
 } from '@wordpress/components';
 import { __, isRTL } from '@wordpress/i18n';
-import { symbol, chevronLeft, chevronRight } from '@wordpress/icons';
+import { chevronLeft, chevronRight } from '@wordpress/icons';
 import { privateApis as routerPrivateApis } from '@wordpress/router';
 import { useViewportMatch } from '@wordpress/compose';
 
 /**
  * Internal dependencies
  */
+import PatternsHeader from './header';
 import Grid from './grid';
 import NoPatterns from './no-patterns';
 import usePatterns from './use-patterns';
 import SidebarButton from '../sidebar-button';
 import useDebouncedInput from '../../utils/use-debounced-input';
 import { unlock } from '../../lock-unlock';
+import { SYNC_TYPES, USER_PATTERN_CATEGORY } from './utils';
 
 const { useLocation, useHistory } = unlock( routerPrivateApis );
 
+const SYNC_FILTERS = {
+	all: __( 'All' ),
+	[ SYNC_TYPES.full ]: __( 'Synced' ),
+	[ SYNC_TYPES.unsynced ]: __( 'Standard' ),
+};
+
 export default function PatternsList( { categoryId, type } ) {
 	const location = useLocation();
 	const history = useHistory();
 	const isMobileViewport = useViewportMatch( 'medium', '<' );
 	const [ filterValue, setFilterValue, delayedFilterValue ] =
 		useDebouncedInput( '' );
+	const deferredFilterValue = useDeferredValue( delayedFilterValue );
 
-	const [ patterns, isResolving ] = usePatterns(
-		type,
-		categoryId,
-		delayedFilterValue
-	);
+	const [ syncFilter, setSyncFilter ] = useState( 'all' );
+	const deferredSyncedFilter = useDeferredValue( syncFilter );
+	const { patterns, isResolving } = usePatterns( type, categoryId, {
```

Comment:
Instead of returning a `synced`/`unsynced` pair, `patterns` is now a filtered array containing all the pattern items.